### ISSUE TITLE
Graphviz Plugin is not working

### ISSUE THREAD

**Issue by omontigny** [bitnami/bitnami-docker-dokuwiki#37]
**Description**

I can't manage to use the following plugin : https://www.dokuwiki.org/plugin:graphviz

**Steps to reproduce the issue:**

1. Install plugin
2. Install graphviz in container (apt-get update, apt-get install graphviz)
3. Test this code in a page
```
<graphviz>
digraph ATN {
 rankdir=LR;
 serv1 [label="SERV1",shape="record"];
 serv3 [label="SERV3",shape="record"];
 serv2 [label="SERV2",shape="record"];

 serv2 -> serv1 [label="PROTO1",dir="right"];
 serv1 -> serv3 [label="PROTO2",dir="right"];

 subgraph cluster_global {
  label="TITLE";
  subgraph cluster_zone1 {
   label="zone1";
   serv1;
  }
  subgraph cluster_zone2 {
   label="zone2";
   serv2;
  }
  subgraph cluster_zone3 {
   label="zone3";
   serv3;
  }
 }
}
</graphviz>
```

**Describe the results you received:**

The page stay blank

**Describe the results you expected:**

We should have a graph pictured (as attachment)

**Additional information you deem important (e.g. issue happens only occasionally):**

**Version**

- Output of `RancherOS and RancherServer version`:

```
Rancher OS : 1.1.1
Rancher Server : 1.6.12
```
- Output of `docker version`:

```
Docker version 17.03.2-ce, build f5ec1e2
```

- Output of `docker info`:

```
Containers: 27
 Running: 26
 Paused: 0
 Stopped: 1
Images: 226
Server Version: 17.03.2-ce
Storage Driver: overlay
 Backing Filesystem: extfs
 Supports d_type: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.67-rancher
Operating System: RancherOS v1.1.1
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 5.829 GiB
Name: xxxxx
ID: xxxxxx
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
Live Restore Enabled: false

```

- Output of `docker-compose version` (if applicable):

```
(paste your output here)
```
<img width="584" alt="graphviz_example" src="https://user-images.githubusercontent.com/17742618/34387459-9c54de28-eb2e-11e7-9167-ccb68e9b7759.png">

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

For additionnal information : 
Despite the fact we can read on the plugin website that it is not compatible with the last dokuwiki version: it is not true because I managed to use the plugin in the michaelsiebertz/dokuwiki:2017-02-19e image (but I have another problem with peristence volume which stay empty. ) 

I still want to use the bitnami image but I don't know how to fix this plugin problem. 

Thank you in advance
Best regards

**Comment by andresbono** [comment]
Hi @omontigny,

**TL;DR;**

```
sed -i "s/<?php/<?php\ndefine\('DOKU_INC','\/opt\/bitnami\/dokuwiki\/');/" /opt/bitnami/dokuwiki/lib/plugins/graphviz/img.php && nami restart apache
```
___
We have reproduced it doing the following:

1. Start our latest Docker image with `docker-compose`:
```
docker-compose up
```

2. Use the extension manager to install the plugin:

http://localhost/doku.php?id=start&do=admin&page=extension&tab=search&q=graphviz

3. Add your snippet of graphviz code in the main page of DokuWiki.

At this point, we can reproduce the issue: a blank page instead of a graph.

4. Check the Apache logs:

```
tail /opt/bitnami/apache/logs/error_log
```
the following line is found:

```
[Thu Dec 28 13:37:26.157654 2017] [:error] [pid 76] [client 172.19.0.1:33790] PHP Warning:  require_once(/bitnami/dokuwiki/lib/plugins/graphviz/../../../inc/init.php): failed to open stream: No such file or directory in /bitnami/dokuwiki/lib/plugins/graphviz/img.php on line 9, referer: http://localhost/doku.php?id=start
```

Checking the `/bitnami/dokuwiki/lib/plugins/graphviz/img.php` file, the root directory of DokuWiki is not correctly calculated (`DOKU_INC`). That's because we use `/bitnami` to locate the application data instead of `/opt/bitnami` so it can be persisted in a volume.

5. Refine `DOKU_INC` constant:

```
sed -i "s/<?php/<?php\ndefine\('DOKU_INC','\/opt\/bitnami\/dokuwiki\/');/" /opt/bitnami/dokuwiki/lib/plugins/graphviz/img.php
```

6. Just in case, restart apache:

```
nami restart apache
```

Note that you don't need to install graphviz in the system:

![screen shot 2017-12-28 at 16 32 48](https://user-images.githubusercontent.com/1057973/34415070-f31c7e00-ebec-11e7-984a-87be7ba409c7.png)


Apart from this workaround, we will try to solve this issue globally by updating our image.

Let us know if this works for you. Hope it helps!

**Comment by omontigny** [comment]
Hi, @andresbono 

Yes it helps !  this workaround solved my problem.
Thank you very much .... 

Is it possible to install graphviz package (/usr/bin/dot)  inside your image ?

Thanks
Best regards

**Comment by andresbono** [comment]
We are glad that you were able to fix the issue! 🎉 

About including graphviz package in images, we try to keep our containers as lightweight as possible and graphviz requires a lot of dependencies that would make the size increase.

If you require it, maybe you want to customize our Dockerfile and include graphviz on it. Keep in mind that you can install it at execution time too but `/usr/bin` directory wont be persisted.

Regards!

**Comment by andresbono** [comment]
BTW, we are going to close this issue as solved, but feel free to keep commenting if you need further help 👍
